### ISSUE TITLE
FX call_function does not work if target is a class member function

### ISSUE THREAD

**Issue by mrzzd** [pytorch/pytorch#50143]
## 🐛 Bug

Currently if one has a statement like `tracer.create_proxy('call_function', f, args, kwargs)` where `f` is a class member function, the generated code does not work correctly.

## To Reproduce

As explained in the description:

```
class A:
    def F(): return 1

tracer.create_proxy('call_function', A.F, tuple(), {})
```

## Expected behavior

The generated code should have `f = module_name.A.f()`. But instead it has `f.module_name.f()`

**Comment by jamesr66a** [comment]
Can you show a broader example of where this is needed? I want to make sure that the rest of the infrastructure is consistent with supporting calling class methods

**Comment by mrzzd** [comment]
> Can you show a broader example of where this is needed? I want to make sure that the rest of the infrastructure is consistent with supporting calling class methods

We have an internal usecase for it. Basically we like to fx-trace a part of the forward function of the module, and the rest is splitted to a separate function of the module, which is called by proxy call_function (using newly introduced @wrap).  Yes, it is a bound method call, and we also need to extend create_arg to support passing "this"  (which is of type nn.Module), but other than that it works fine.

**Comment by jamesr66a** [comment]
OK so the requirements are basically:
* You want a callee method of forward() on the same class to be accessible from the final GraphModule interface
* The implementation of this callee method should be traced through

I think we should provide an API for this rather than using wrap, which is meant for free functions. We might have something like:

```
class Foo(torch.nn.Module):
  @torch.fx.traced_method
  def callee(self, x):
    return torch.relu(x)

  def forward(self, x):
    return self.callee(x) + 3.0
```

Then the traced forward call would have a `call_method` node with `self` as the target.

Does this sound like it matches your use case?

**Comment by mrzzd** [comment]
> Does this sound like it matches your use case?

Yes, that's what we need. Just small point that `self` is an arg here not a target.
